src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -7472,6 +7472,34 @@ static void Main()
 
             CleanupAllGeneratedFiles(src.Path);
         }
+
+        [Fact]
+        public void ParseAnalyzerDependencies()
+        {
+            var parsedArgs = CSharpCommandLineParser.Default.Parse(new string[] { @"/analyzerdependency:foo.dll", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify();
+            Assert.Equal(1, parsedArgs.AnalyzerDependencies.Length);
+            Assert.Equal("foo.dll", parsedArgs.AnalyzerDependencies[0].FilePath);
+
+            parsedArgs = CSharpCommandLineParser.Default.Parse(new string[] { "/analyzerdependency:\"foo.dll\"", "a.cs" }, _baseDirectory);
```

Comment:
\"\" won't trigger the null or empty check, as a string consisting of two double-quotes isn't null or empty. That check is just there to make sure the switch has a value of some sort; further validation will be done later.